src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -2446,8 +2492,15 @@ export class StudyViewPageStore {
         return { ...this.filters, sampleIdentifiersSet };
     }
 
-    public getGeneFiltersByUniqueKey(uniqueKey: string) {
-        return toJS(this._geneFilterSet.get(uniqueKey)) || [];
+    public getGeneFiltersByUniqueKey(uniqueKey: string): string[][] {
+        const filters = _.map(this._geneFilterSet.get(uniqueKey), fs =>
+            _.map(fs, f =>
+                f.alterations.length > 0
+                    ? `${f.hugoGeneSymbol}:${f.alterations[0]}`
```

Comment:
comment that explains the composition of this key.  like, why first alteration is always used

Reply:
The logic in this loop is present in the util function _geneFilterQueryToOql_. Replaced this logic with a call to this method and considered this sufficient documentation for this piece of code:

```
    public getGeneFiltersByUniqueKey(uniqueKey: string): string[][] {
        const filters = _.map(this._geneFilterSet.get(uniqueKey), filterSet =>
            _.map(filterSet, geneFilterQueryToOql)
        );
        return toJS(filters);
    }
```
The key is not composed of all alterations.